### ISSUE TITLE
SplitColorComposite generates no output

### ISSUE THREAD

**Issue by PedroVenancio** [alexbruy/processing-whitebox#17]
Hi @alexbruy 

I'm trying SplitColourComposite algorithm in QGIS 3.6 (Windows), but I'm not getting any output:

```
Processing algorithm…
Algorithm 'SplitColourComposite' starting…
Input parameters:
{ 'input' : 'D:/orto_teste.tif', 'output' : 'D:/orto_teste_split_2.tif' }

WhiteBoxTools command:
"C:\OSGeo4W64\WhiteboxTools_win_amd64\WBT\whitebox_tools.exe" --run=SplitColourComposite --input="D:\orto_teste.tif" --output="D:\orto_teste_split_2.tif" -v
WhiteBoxTools command output:
***********************************

* Welcome to SplitColourComposite *

***********************************

Reading data...

Saving red image...

Output file written

Saving green image...

Output file written

Saving blue image...

Output file written

Elapsed Time (excluding I/O): 0.67s

Execution completed in 0.43 seconds
Results:
{'output': <QgsProcessingOutputLayerDefinition {'sink':D:/orto_teste_split_2.tif, 'createOptions': {'fileEncoding': 'ISO-8859-1'}}>}

Loading resulting layers
The following layers were not correctly generated.<ul><li>D:/orto_teste_split_2.tif</li></ul>You can check the 'Log Messages Panel' in QGIS main window to find more information about the execution of the algorithm.
```

Running exactly the same WhiteBoxTools command, from the command line, I get the correct result:

```
C:\>"C:\OSGeo4W64\WhiteboxTools_win_amd64\WBT\whitebox_tools.exe" --run=SplitColourComposite --input="D:\orto_teste.tif" --output="D:\orto_teste_split_2.tif" -v
***********************************
* Welcome to SplitColourComposite *
***********************************
Reading data...
Progress: 0%
Progress: 1%
Progress: 2%
Progress: 3%
Progress: 4%
Progress: 5%
Progress: 6%
Progress: 7%
Progress: 8%
Progress: 9%
Progress: 10%
Progress: 11%
Progress: 12%
Progress: 13%
Progress: 14%
Progress: 15%
Progress: 16%
Progress: 17%
Progress: 18%
Progress: 19%
Progress: 20%
Progress: 21%
Progress: 22%
Progress: 23%
Progress: 24%
Progress: 25%
Progress: 26%
Progress: 27%
Progress: 28%
Progress: 29%
Progress: 30%
Progress: 31%
Progress: 32%
Progress: 33%
Progress: 34%
Progress: 35%
Progress: 36%
Progress: 37%
Progress: 38%
Progress: 39%
Progress: 40%
Progress: 41%
Progress: 42%
Progress: 43%
Progress: 44%
Progress: 45%
Progress: 46%
Progress: 47%
Progress: 48%
Progress: 49%
Progress: 50%
Progress: 51%
Progress: 52%
Progress: 53%
Progress: 54%
Progress: 55%
Progress: 56%
Progress: 57%
Progress: 58%
Progress: 59%
Progress: 60%
Progress: 61%
Progress: 62%
Progress: 63%
Progress: 64%
Progress: 65%
Progress: 66%
Progress: 67%
Progress: 68%
Progress: 69%
Progress: 70%
Progress: 71%
Progress: 72%
Progress: 73%
Progress: 74%
Progress: 75%
Progress: 76%
Progress: 77%
Progress: 78%
Progress: 79%
Progress: 80%
Progress: 81%
Progress: 82%
Progress: 83%
Progress: 84%
Progress: 85%
Progress: 86%
Progress: 87%
Progress: 88%
Progress: 89%
Progress: 90%
Progress: 91%
Progress: 92%
Progress: 93%
Progress: 94%
Progress: 95%
Progress: 96%
Progress: 97%
Progress: 98%
Progress: 99%
Progress: 100%
Saving red image...
Output file written
Saving green image...
Output file written
Saving blue image...
Output file written
Elapsed Time (excluding I/O): 0.80s
```

**Comment by jblindsay** [comment]
This should probably be reported on the [WhiteboxTools Github issues](https://github.com/jblindsay/whitebox-tools) page instead as this repository is only for the QGIS WhiteboxTools frontend, and your issue seems to be with the underlying tool. Would you mind reporting it there instead, along with information about what version of WhiteboxTools you are using and an indication of the data type/format, etc.? I just tested the tool with some Landsat 8 data and it worked okay on my system.

**Comment by PedroVenancio** [comment]
Hi @jblindsay 

It does not seem a Whitebox problem, since it works launching the algorithm from the command line. 

I leave a sample dataset (tif) to test.


[orto_teste.zip](https://github.com/alexbruy/processing-whitebox/files/2906602/orto_teste.zip)

**Comment by alexbruy** [comment]
@PedroVenancio actually this is not a bug, rather a mix of the Processing and WhiteboxTools features. This tool produces multiple files, not single file. And output file names are constructed from the passed file name and some suffixes. So while outputs are generated, Processing still can not find a file with expected file name.

Seems I need to think about implementing additional extensions for such tools.

**Comment by PedroVenancio** [comment]
Hi @alexbruy 

I didn't noticed that the output was being created! 

This is a tricky one... hope you can find a solution!

Just another question, not directly related to this topic, but the link to the manual is not being showed in Processing. Is there something that can be done?

![link_to_manual](https://user-images.githubusercontent.com/1331581/53482029-3d451400-3a76-11e9-84db-120aec7bfea0.jpg)

Thank you very much @alexbruy !

**Comment by alexbruy** [comment]
@PedroVenancio why it should be there? There is a help button which opens manual.

**Comment by PedroVenancio** [comment]
> @PedroVenancio why it should be there? There is a help button which opens manual.

:) Forget, my fault!

Thanks @alexbruy !!

**Comment by jblindsay** [comment]
@alexbruy I've just committed an updated version of the tool to the WhiteboxTools repo that requires all three, red, green, and blue, output files specified separately. I'm hoping to release 0.15.1 sometime in the next couple of weeks (after I get some grad defenses completed) and it will be available publicly then. Hopefully this change will be visible to each of the WhiteboxTools front-ends and they will update as well.

**Comment by alexbruy** [comment]
Fixed in the WhiteboxTools 0.16 and provider plugin version 0.12
